string.rb

```diff
@@ -1,556 +1,556 @@
-#    'foo'[-2, 2]   # => "oo"
+#   'foo'[-2..-1]   # => "oo"
+#   'foo'[-2, 2]    # => "oo"
 #
-#    'foo'[-2..197] # => "oo"
-#    'foo'[-2, 200] # => "oo"
+#   'foo'[-2..197]  # => "oo"
+#   'foo'[-2, 200]  # => "oo"
 #
-#  <b><tt>string[regexp, capture = 0]</tt></b>
+# <b><tt>string[regexp, capture = 0]</tt></b>
 #
-#  When the Regexp argument +regexp+ is given,
-#  and the +capture+ argument is <tt>0</tt>,
-#  the slice is the first matching substring found in +self+:
+# When the Regexp argument +regexp+ is given,
+# and the +capture+ argument is <tt>0</tt>,
+# the slice is the first matching substring found in +self+:
 #
-#    'foo'[/o/] # => "o"
-#    'foo'[/x/] # => nil
-#    s = 'hello there'
-#    s[/[aeiou](.)\1/] # => "ell"
-#    s[/[aeiou](.)\1/, 0] # => "ell"
+#   'foo'[/o/]                # => "o"
+#   'foo'[/x/]                # => nil
+#   s = 'hello there'
+#   s[/[aeiou](.)\1/]        # => "ell"
+#   s[/[aeiou](.)\1/, 0]     # => "ell"
 #
-#  If argument +capture+ is given and not <tt>0</tt>,
-#  it should be either an capture group index (integer)
-#  or a capture group name (string or symbol);
-#  the slice is the specified capture (see Regexp@Groups+and+Captures):
+# If the argument +capture+ is provided and not <tt>0</tt>,
+# it should be either a capture group index (integer)
+# or a capture group name (String or Symbol);
+# the slice is the specified capture (see Regexp@Groups and Captures):
 #
-#    s = 'hello there'
-#    s[/[aeiou](.)\1/, 1] # => "l"
-#    s[/(?<vowel>[aeiou])(?<non_vowel>[^aeiou])/, "non_vowel"] # => "l"
-#    s[/(?<vowel>[aeiou])(?<non_vowel>[^aeiou])/, :vowel] # => "e"
+#   s = 'hello there'
+#   s[/[aeiou](.)\1/, 1] # => "l"
+#   s[/(?<vowel>[aeiou])(?<non_vowel>[^aeiou])/, "non_vowel"] # => "l"
+#   s[/(?<vowel>[aeiou])(?<non_vowel>[^aeiou])/, :vowel]      # => "e"
 #
-#  If an invalid capture group index is given, there is no slice.
-#  If an invalid capture group name is given, +IndexError+ is raised.
+# If an invalid capture group index is given, there is no slice.
+# If an invalid capture group name is given, +IndexError+ is raised.
 #
-#  <b><tt>string[substring]</tt></b>
+# <b><tt>string[substring]</tt></b>
 #
-#  When the single +String+ argument +substring+ is given,
-#  returns the substring from +self+ if found, otherwise +nil+:
+# When the single +String+ argument +substring+ is given,
+# it returns the substring from +self+ if found, otherwise +nil+:
 #
-#    'foo'['oo'] # => "oo"
-#    'foo'['xx'] # => nil
+#   'foo'['oo'] # => "oo"
+#   'foo'['xx'] # => nil
 #
-#  == What's Here
+# == What's Here
 #
-#  First, what's elsewhere. \Class +String+:
+# First, what's elsewhere. \Class +String+:
 #
-#  - Inherits from {class Object}[rdoc-ref:Object@What-27s+Here].
-#  - Includes {module Comparable}[rdoc-ref:Comparable@What-27s+Here].
+# - Inherits the {Object class}[rdoc-ref:Object@What-27s+Here].
+# - Includes the {Comparable module}[rdoc-ref:Comparable@What-27s+Here].
```

Comment:
FWIW, if in the future we can enable the [`--embed-mixins`](https://github.com/ruby/rdoc/pull/842) flag added by @flavorjones in `ruby/ruby`, these methods will be added to the `String` class's doc directly.